src/hooks/use-show-value-outside.ts

```diff
@@ -0,0 +1,80 @@
+import { useEffect, useRef } from "react"
+import { isEmpty } from "ramda"
+
+import { ChartData, DygraphData } from "domains/chart/chart-types"
+import { Attributes } from "domains/chart/utils/transformDataAttributes"
+import { ChartLibraryConfig } from "domains/chart/utils/chartLibrariesSettings"
+
+
+interface UseShowValueOutsideArgument {
+  attributes: Attributes
+  chartData: ChartData
+  chartSettings: ChartLibraryConfig
+  hoveredRow: number
+  legendFormatValue: ((v: number | string | null) => number | string)
+  showUndefined: boolean
+}
+
+// example of the attribute:
+// show-value-of-iowait-at: "system.cpu.iowait.1"
+
+export const useShowValueOutside = ({
+  attributes,
+  chartData,
+  chartSettings,
+  hoveredRow,
+  legendFormatValue,
+  showUndefined,
+}: UseShowValueOutsideArgument) => {
+  // a ref to store found elements, just once per lifetime of component
+  const showValueAttributesNodes = useRef<(HTMLElement | null)[]>([])
+
+  // find the nodes that will have populated values
+  useEffect(() => {
+    const { showValueOf } = attributes
+    // showValueOf will be undefined if not used, but additional isEmpty check can prevent
+    // regression performance issue in the future
+    if (!showValueOf || isEmpty(showValueOf)) {
+      return
+    }
+    const dimensionNames = chartData.dimension_names
+    const dimensionIds = chartData.dimension_ids
+    dimensionNames.forEach((dimensionName, i) => {
+      const userElementId = showValueOf[`show-value-of-${dimensionName.toLowerCase()}`]
```

Comment:
Could be an external util, maybe this level of detail how we call the key isn't so important.

Reply:
do you mean that `show-value-of-${dimension}-at` is too descriptive? This is the exact name of an attribute from custom dashboards

Reply:
Well, technically of course this is needed, I'm saying that probably it's not required to see the exact string templates when reading the code there.